internal/grpc/proto/kong/admin/model/v1/vault.proto

```diff
@@ -0,0 +1,19 @@
+syntax = "proto3";
+
+package kong.admin.model.v1;
+
+import "google/api/field_behavior.proto";
+import "google/protobuf/struct.proto";
+
+option go_package = "github.com/kong/koko/internal/gen/grpc/kong/admin/model/v1;v1";
+
+message Vault {
+  string id = 1;
+  int32 created_at = 2;
+  int32 updated_at = 3;
+  string prefix = 4 [(google.api.field_behavior) = REQUIRED];
+  string name = 5 [(google.api.field_behavior) = REQUIRED];
+  string description = 6;
+  google.protobuf.Struct config = 7;
```

Comment:
Can we get this properly typed instead?
```protobuf
message Vault {
  message GCPSecretManager {
    string project_id = 1;
  }
  message AWSSecretsManager {
    string region = 1;
  }
  message HashiCorpVault {
    string host = 1;
    int32 port = 2;
    string protocol = 3;
    string mount = 4;
    string kv = 5;
    string token = 6;
  }
  message EnvironmentVariables {
    string prefix = 1;
  }

  oneof config {
    GCPSecretManager gcp_secret_manager = 7;
    AWSSecretsManager aws_secrets_manager = 8;
    HashiCorpVault hashicorp_vault = 9;
    EnvironmentVariables env = 10;
  }
}
```

Reply:
In order to match the existing api, I think we'll have to go with something like:

```protobuf
message Vault {
  string id = 1;
  int32 created_at = 2;
  int32 updated_at = 3;
  string prefix = 4 [(google.api.field_behavior) = REQUIRED];
  string name = 5 [(google.api.field_behavior) = REQUIRED];
  string description = 6;

  message Config {
    string prefix = 1;
    string region = 2;
    string project_id = 3;
    string host = 4;
    int32 port = 5;
    string protocol = 6;
    string mount = 7;
    string kv = 8;
    string token = 9;
  }

  Config config =  7;
  repeated string tags = 8;
}
```